staticcheck/lint.go

```diff
@@ -117,6 +117,51 @@ func checkValidHostPort(arg int) CallCheck {
 	}
 }
 
+func checkNonOverlappingDstSrc(dstArg, srcArg int) CallCheck {
+	return func(call *Call) {
```

Comment:
We have multiple places that check for and ignore `nil`. It'd probably make sense to check if either src or dst are nil and return early.